modules/framework/galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/TestMethodWrapper.java

```diff
@@ -79,6 +79,12 @@ public void invoke(
 
                 runBeforeMethods(managers, testClassObject, testClassWrapper);
 
+                // TODO After this point, should we invoke after methods even if the test method fails with an exception ?
+                // Seems that an exception will skip lots of things otherwise.
+                // Failures like an exception being thrown from the testcase method itself don't get thrown up to this point, only if 
+                // there is a bad framework issue such as the dss being unavailable or something. In which case all bets are probably off anyway.
+                // Managers will be invited to clean up even if we throw an exception out of this method.
```

Comment:
I would assume that the after methods would be invoked to allow some cleanup to take place if possible (e.g. like resetting a terminal screen), but yes, with wider framework issues, exiting out and letting the managers clean up sounds fine.

I think this is OK.